File: computation numeric/iterasi/java/Iterasi.java
package main;

public class Iterasi {

	public static void main(String[] args) {
		double x=0;
		int i=0;
		while (i !=10) {
			double tmp = hasilFX(x);
			System.out.println("x"+i+": "+x+"\tFx\t: "+tmp);
			x= tmp;
			
			i++;
		}	
	}
	public static double hasilFX(double x) {
		double fx = (Math.exp(-x));
		return fx;
		
	}
}

File: computation numeric/iterasi/python/iterasi.py
import math

def f(x):
    fx = math.exp(-x)
    return fx

def fixed_point(max_step = 20, tol = 0.0001):
    
    p = 0
    
    loopcounter = 1
    
    for loopcounter in range(max_step):
        
        oldP = f(p)
        
        print('iterasi ke ', int(loopcounter), ' xn ', p, 'e^(xn)', oldP)
        
        if (abs(p - oldP) <= tol):
            print('root value is ', oldP, 'in steps ', int(loopcounter))
            return
            
        p = oldP

fixed_point()